### PR TITLE
Swagger apiview improvement and bug fix

### PR DESCRIPTION
**Pull request by ruowan**
- Fix missing first property for definition
- Fix wrong "required" property in xMsParameterizedHost parameters by resolving parameter reference
- Show enum type for definition
- Support resolve reference from common types

Issue: 

- https://github.com/Azure/azure-sdk-tools/issues/4462
- https://github.com/Azure/azure-sdk-tools/issues/4463
- https://github.com/Azure/azure-sdk-tools/issues/4461

Link: https://apiviewstaging.azurewebsites.net/Assemblies/Review/4efa16b197314f9896478eeaa09cc8d2

### CODE REVIEW

**Comment by ruowan** [comment]
https://github.com/Azure/azure-sdk-tools/issues/4462

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/SwaggerApiParser.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <TargetFramework>net6.0</TargetFramework>
     <RootNamespace>swagger_api_parser</RootNamespace>
     <LangVersion>Latest</LangVersion>
-    <VersionPrefix>1.0.0</VersionPrefix>
+    <VersionPrefix>1.0.1</VersionPrefix>
```
Comment (maririos) [inline tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/SwaggerApiParser.csproj:10]: Could we add a changelog file to the folder? i.e https://github.com/Azure/azure-sdk-tools/blob/main/tools/apiview/emitters/cadl-apiview/CHANGELOG.md

Comment (ruowan) [reply tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/SwaggerApiParser.csproj:10]: Good. Added changelog

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs [hunk 2]
```diff
@@ -79,6 +79,10 @@ static async Task HandleGenerateCodeFile(IEnumerable<string> swaggers, string ou
             
             
             SwaggerApiViewRoot root = new SwaggerApiViewRoot(packageName, packageName);
```
Comment (praveenkuttappan) [inline tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs:81]: Unrelated to this PR. But why are we passing same variable twice to SwaggerApiViewroot?
`packageName`

**Review by praveenkuttappan — APPROVED** [review]

File: tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs (hunk 2 above)
Comment (ruowan) [reply tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/Program.cs:81]: Cood catch. I will check it.

**Pull request merged**